mautic/mautic
# Point email action improvements

**Please be sure you are submitting this against the _staging_ branch.**

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | 
| New feature? | 
| Automated tests included? |
| Related user documentation PR URL | 
| Related developer documentation PR URL | 
| Issues addressed (#s or URLs) | Resolves https://github.com/mautic/mautic/issues/7320
| BC breaks? | 
| Deprecations? | 

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:
This PR try figure out trigger points for each entity in points action reported in this issue https://github.com/mautic/mautic/issues/7320

- added support for events system (eventName instead of callback, callback should be depraced and refactored in future)
- added  new column **internal_id** to point_lead_action_log table - ID or string - in thu future we can store URL for page hit and then we can run points action on each visits
- added new option to email open point action - categories limit and execute one time/each email

![image](https://user-images.githubusercontent.com/462477/54570310-60604500-49de-11e9-9f30-677895652f94.png)

#### Steps to test this PR:
1. Load up [this PR](https://mautibox.com)
2. Create point actions - opens email type
3. Test step by step with different point actions
- Allow repeat the action - should change points everytime
- Execute only: Once - should change points just one time 
- Execute only: For each email - should change points with every email ID - that means If I will create point action just for 3 emails, and all 3 emails my contact open , contact point score adjust point score each time
4. Test also categories filter
 

#### List deprecations along with the new alternative:
1. callback depraced on triggerAction https://github.com/mautic/mautic/blob/0e500aee172676a591341e090873c4120d796a4d/app/bundles/EmailBundle/EventListener/PointSubscriber.php#L82
Use eventName like in other parts of Mautic.


<a href="https://gitpod.io/#https://github.com/mautic/mautic/pull/7330"><img src="https://gitpod.io/button/open-in-gitpod.svg"/></a>
